### DISCUSSION TITLE
Add basic multiplayer room filters in lounge screen

### DISCUSSION THREAD

**Discussion by TheFantasticLoki** [ppy/osu#17498] [Ideas]
I'd like to see the addition of some basic filters to the multiplayer lobby select screen such as hiding password protected lobbies, games with friends, and owned beatmaps. 
The majority of lobbies in lazer are currently password protected and therefore can't be joined by randoms so why should we have to manually sort through them instead of just hiding them?

**Comment by ggliv** [comment]
I'm [looking to add at least the locked room filter](https://github.com/ggliv/osu/compare/master..locked-room-filter), but I wanted to check on the design before opening a pull request. My changes use an `OsuTabControlCheckbox` to toggle the hidden room filter, but I don't think that's proper given that it is used without an associated `OsuTabControl`. The [osu! Figma](https://www.figma.com/file/XJCRjfvTlVIwZBWoAz5kB2/Client%2FPlaylist-Multiplayer) doesn't seem to have a design laid out.  Do any team members have opinions or directions on how this should look, or is this good enough for a temporary solution?

![image](https://user-images.githubusercontent.com/47010459/175141007-14976092-e634-4e02-ab79-cffcc7e41b14.png)
![image](https://user-images.githubusercontent.com/47010459/175141154-f9c18f57-95e7-4c85-99c8-3b1003f7343b.png)

**Reply by frenzibyte** [reply]
The line at the bottom of the checkbox should be hidden, and the checkbox may need to get a bit scaled up to match text size with dropdowns, but I think it looks fine to me? Will wait for a second opinion.

**Reply by bdach** [reply]
The checkbox does not fit at all. If you ask me, this should be a dropdown with public/private being the available options.

**Reply by peppy** [reply]
Any of the above work for me. Whatever is the easiest to implement (as it will definitely be replaced visually when we get to refreshing that whole area, so don't spend too long making it perfect).

**Comment by ggliv** [comment]
#18821 will handle the password protected filter

**Comment by godlikeswan** [comment]
The public/private filter is already implemented. The "show the rooms where your friends are" would be nice, probably need a new discussion started.
